### PR TITLE
list mode selection should be keep through navigation

### LINKED ISSUE #311: list mode selection should be keep through navigation
**Describe the bug**

Session schedules can be shown in timetable mode or list mode.
This selection should be kept through navigation like moving to detail and back, but it will be reset to timetable mode after navigation.

**To Reproduce**

1. show timetable 
2. switch to list mode by clicking the calendar icon
3. move to session detail
4. click the back button
5. timetable is shown (should be list mode)

**Expected behavior**

Session schedules are shown in list mode after moving to detail and back.

**Screenshots**

https://user-images.githubusercontent.com/455504/189278274-a33bfcca-d037-49df-979f-da0da647dbfc.mp4

### PR DESCRIPTION
**Pull request by sudachi808**
## Issue
- close #311

## Overview (Required)
- Provide mode selection with ViewModel.
- Save mode selection on DataStore.

## Screenshot

https://user-images.githubusercontent.com/26338889/189364426-26569af2-c782-4c49-a978-f556e84e3276.mp4

### CODE REVIEW

File: core-model/src/commonMain/kotlin/io/github/droidkaigi/confsched2022/model/SessionsRepository.kt [hunk 1]
```diff
@@ -13,4 +13,7 @@ interface SessionsRepository {
 
     suspend fun refresh()
     suspend fun setFavorite(sessionId: TimetableItemId, favorite: Boolean)
+
+    fun isTimetableModeFlow(): Flow<Boolean>
+    suspend fun setIsTimetableMode(isTimetableMode: Boolean)
```
Comment (sudachi808) [inline core-model/src/commonMain/kotlin/io/github/droidkaigi/confsched2022/model/SessionsRepository.kt:18]: better to separate as any other repository class?

File: core-data/src/androidMain/kotlin/io/github/droidkaigi/confsched2022/data/sessions/di/SessionDataModule.kt [hunk 2]
```diff
@@ -27,7 +27,7 @@ class SessionDataModule {
         return DataSessionsRepository(
             sessionsApi = sessionsApi,
             sessionsDao = sessionsDao,
-            favoriteSessionsDataStore = settingsDatastore
+            settingsDatastore = settingsDatastore
```
Comment (takahirom) [inline core-data/src/androidMain/kotlin/io/github/droidkaigi/confsched2022/data/sessions/di/SessionDataModule.kt:30]: 👍

**Comment by takahirom** [comment]
I didn't mean to hold the state in datastore. 👀 I wanted you just to hold it in ViewModel. 🙇

File: feature-sessions/src/main/java/io/github/droidkaigi/confsched2022/feature/sessions/SessionsViewModel.kt [hunk 3]
```diff
@@ -52,6 +52,8 @@ class SessionsViewModel @Inject constructor(
         }
         .asResult()
 
+    private val isTimetableMode = mutableStateOf(true)
```
Comment (sudachi808) [inline feature-sessions/src/main/java/io/github/droidkaigi/confsched2022/feature/sessions/SessionsViewModel.kt:55]: store mode selection within ViewModel.

File: feature-sessions/src/main/java/io/github/droidkaigi/confsched2022/feature/sessions/SessionsViewModel.kt [hunk 4]
```diff
@@ -88,7 +87,7 @@ class SessionsViewModel @Inject constructor(
 
     fun onTimetableModeToggle(currentIsTimetableMode: Boolean) {
         viewModelScope.launch {
-            sessionsRepository.setIsTimetableMode(!currentIsTimetableMode)
+            isTimetableMode.value = !currentIsTimetableMode
```
Comment (sudachi808) [inline feature-sessions/src/main/java/io/github/droidkaigi/confsched2022/feature/sessions/SessionsViewModel.kt:90]: updating variable that belongs to ViewModel.

**Review by sudachi808 — COMMENTED** [review]
modified to hold mode selection in ViewModel.

**Review by sudachi808 — COMMENTED** [review]
Sorry... I'd inserted unnecessary blank line by mistake.
It may causes CI build error.

**Review by sudachi808 — COMMENTED** [review]
Very sorry again...🙇 

In this time,  I executed `./gradlew spotlessKotlinCheck` and confirmed successful output.

**Review by takahirom — APPROVED** [review]
LGTM! Thanks for your cooperation🙏

**Pull request merged**